sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/CatalogFileIndex.scala

```diff
@@ -56,11 +60,17 @@ class CatalogFileIndex(
 
   override def listFiles(
       partitionFilters: Seq[Expression], dataFilters: Seq[Expression]): Seq[PartitionDirectory] = {
-    filterPartitions(partitionFilters).listFiles(Nil, dataFilters)
+    val (partitions, phase) = createPhaseSummary {
```

Comment:
how hard it is to do it at caller side?

Reply:
Thank you Wenchen for your review.
Not so hard, but need more code changes on caller class. I choose this just for little change and code clean. The current approach need call `createPhaseSummary`  here and `InmemoryFileIndex.refresh0`, if changes to caller side need tracker more place call listFiles and `new InMemoryFileIndex`.
Another problem is how to pass the phase to ScanExec. If we agree still keep it in FileIndex, need a set function of the phaseSummary for caller side.